onnx_caffe2/backend.py

```diff
@@ -597,18 +657,18 @@ def prepare(cls, model, device='CPU', **kwargs):
 
     @classmethod
     # TODO: This method needs a refactor for clarity
-    def _onnx_node_to_caffe2_op(cls, node_def, opset_version):
+    def _onnx_node_to_caffe2_op(cls, onnx_model, model_helper, node_def, opset_version):
         if node_def.op_type in cls._special_operators:
             translator = getattr(cls, cls._special_operators[node_def.op_type])
         else:
             translator = cls._common_onnx_node_to_caffe2_op
-        ops = translator(OnnxNode(node_def), opset_version)
+        ops = translator(onnx_model, model_helper, OnnxNode(node_def), opset_version)
         if not isinstance(ops, collections.Iterable):
             ops = [ops]
         return ops
 
     @classmethod
-    def _common_onnx_node_to_caffe2_op(cls, onnx_node, opset_version):
+    def _common_onnx_node_to_caffe2_op(cls, onnx_model, model_helper, onnx_node, opset_version):
```

Comment:
this function should NOT take `model_helper` as there are call sites which manipulate the return values (e.g. appending the outputs). If the implementation actually appends to model_helper it'd break

Reply:
I believe it's safe in practice. All the call sites which mutate the output do so after calling `_common_onnx_node_to_caffe2_op` on a particular node type (e.g. Slice). The only op that actually uses model_helper is LSTM, and no one tries to mutate the outputs after operating on an LSTM.

That said, yeah it's still a bit ugly. But what's the alternative? I need model_helper.

Reply:
I was originally suggesting to make only _common_onnx_node_to_caffe2_op not receive model_helper and every other translation function to append to model_helper directly instead of returning a list of ops.

You current approach seems to work too, so let's go with it